mapper-pojo/src/main/java/org/hibernate/search/v6poc/entity/pojo/dirtiness/building/impl/PojoImplicitReindexingResolverOriginalTypeNodeBuilder.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Hibernate Search, full-text search for your domain model
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.search.v6poc.entity.pojo.dirtiness.building.impl;
+
+import java.util.Collection;
+import java.util.Optional;
+
+import org.hibernate.search.v6poc.entity.pojo.dirtiness.impl.PojoImplicitReindexingResolver;
+import org.hibernate.search.v6poc.entity.pojo.dirtiness.impl.PojoImplicitReindexingResolverOriginalTypeNode;
+import org.hibernate.search.v6poc.entity.pojo.dirtiness.impl.PojoImplicitReindexingResolverPropertyNode;
+import org.hibernate.search.v6poc.entity.pojo.model.path.impl.BoundPojoModelPathTypeNode;
+
+class PojoImplicitReindexingResolverOriginalTypeNodeBuilder<T>
+		extends AbstractPojoImplicitReindexingResolverTypeNodeBuilder<T, T> {
+
+	PojoImplicitReindexingResolverOriginalTypeNodeBuilder(BoundPojoModelPathTypeNode<T> modelPath,
+			PojoImplicitReindexingResolverBuildingHelper buildingHelper) {
+		super( modelPath, buildingHelper );
+	}
+
+	@Override
+	public Optional<PojoImplicitReindexingResolver<T>> build() {
+		Collection<PojoImplicitReindexingResolverPropertyNode<? super T, ?>> immutablePropertyNodes =
+				buildPropertyNodes();
+
+		boolean markForReindexing = isMarkForReindexing();
```

Comment:
`shouldMarkForReindexing` it's not pretty but it would be better I think. The current naming is kind of confusing. I know I didn't understand it correctly at a quick glance.